### PR TITLE
Jetpack: Remove obsolete isJetpackMinimumVersion checks

### PR DESCRIPTION
**Pull request by tyxla**
Currently we use `isJetpackMinimumVersion` in many locations, to display a feature or functionality only for a certain Jetpack version or newer. However, Jetpack's official support policy is to support the current - 1 version. Any older versions aren't supported. This means we can remove any checks that check for 7.9-only or older versions of Jetpack. This basically means removing all of the `isJetpackMinimumVersion` usages.

We keep the `isJetpackMinimumVersion` selector though, as it might be useful for future features.

This cleanup helps in removing some dead unnecessary code and keeping the codebase cleaner.

#### Changes proposed in this Pull Request

* Jetpack: Remove obsolete `isJetpackMinimumVersion` checks.

#### Testing instructions
Test the following screens and flows and verify they work like they did before:
* Simple payments dialog in classic editor (Jetpack and WP.com site).
* Memberships in the Simple Payments dialog in classic editor.
* Comments - navigation - newest / oldest sorting functionality.
* Comments - comment tree and comment nesting.
* Comments - comment inline editing.
* Comments - main screen.
* Tools -> Earn -> Payments - recurring payments.
* Tools -> Marketing -> Sharing Buttons -> Options card settings.
* Tools -> Marketing -> Connections -> Mailchimp connection.
* Tools -> Marketing - The "Get social, and share your blog posts where the people are" card.
* Plans - Jetpack backup purchase card for Jetpack sites.
* Plugins -> Upload (http://calypso.localhost:3000/plugins/upload/:site), test with single site and multisite (multisite should suggest going to wp-admin).
* Pages -> Edit Page (classic editor) -> Sidebar -> Page Attributes, selecting a parent page.
* Posts -> Edit Post (classic editor) -> Sidebar -> Categories & Tags, displaying categories with the current category selected and tags inputted.
* Posts -> Edit Post (classic editor) -> Sidebar - test everything on a Jetpack and WP.com site.
* Posts -> Edit Post (classic editor) -> Sidebar -> Categories & Tags, selecting a category and inputting tags.
* Not removing the selector because it will be necessary as we release new features.

### CODE REVIEW

**Review by delawski — APPROVED** [review]
This is grand, @tyxla! 😀

The code looks good. I've tested locally almost all cases you listed. I just wasn't able to preview the following case:
> * Memberships in the Simple Payments dialog in classic editor.

Besides that, everything worked as expected, I haven't noticed anything suspicious.

I think it's ready to be merged 👍

**Comment by mmtr** [comment]
> However, Jetpack's official support policy is to support the current - 1 version. Any older versions aren't supported.

> This means we can remove any checks that check for 7.9-only or older versions of Jetpack

I don't think these 2 statements are equivalent. Do we have any policy in Calypso establishing the number of Jetpack versions that should be supported?

**Comment by tyxla** [comment]
> I don't think these 2 statements are equivalent. Do we have any policy in Calypso establishing the number of Jetpack versions that should be supported?

It's a fair question, but I think they are equivalent, and let me explain why. Since Jetpack is required for supporting the features in question, if Jetpack doesn't support a certain feature, it doesn't make sense for Calypso to support it either, because it might still be failing due to lack of Jetpack support. This means that the policies should be the same as it comes to Jetpack version support, regardless of which platform we're developing in. Does that make sense?

**Comment by mmtr** [comment]
> if Jetpack doesn't support a certain feature, it doesn't make sense for Calypso to support it either, because it might still be failing due to lack of Jetpack support. 

Yeah, that makes sense but I'm thinking in the opposite scenario. Let's say a Jetpack feature was introduced recently, like 3 releases ago. By supporting only the 2 latest releases of Jetpack, we will assume that the feature is available without checking the version, but that feature in Calypso won't work for users who didn't update Jetpack in the last 2 months (since Jetpack releases are monthly).

**Comment by tyxla** [comment]
> By supporting only the 2 latest releases of Jetpack, we will assume that the feature is available without checking the version, but that feature in Calypso won't work for users who didn't update Jetpack in the last 2 months (since Jetpack releases are monthly).

Yes, and that's expected. We don't want users to be running unsupported versions of Jetpack, and we don't want to encourage them to keep unsupported versions of Jetpack. By removing those checks, we remove that encouragement, and if something is broken for them, it would be an ideal occasion to update to a supported version.

**Comment by mmtr** [comment]
> We don't want users to be running unsupported versions of Jetpack

Gotcha. I thought that the current - 1 version support was referring only to the WP Core versions supported by Jetpack when developing a feature and not to the Jetpack versions supported by Automattic (i.e. HEs). Thanks for clarifying!

**Comment by tyxla** [comment]
> Gotcha. I thought that the current - 1 version support was referring only to the WP Core versions supported by Jetpack when developing a feature and not to the Jetpack versions supported by Automattic (i.e. HEs). 

Yeah, this is true for WP core versions, but it is also a valid policy for Jetpack releases themselves, and we apply it when doing version checks on the Jetpack side, too. And I think it only makes sense to be consistent and follow the same policy in Calypso.

In any case, I'll cc @jeherve to confirm this on behalf of @Automattic/jetpack-crew, as we discussed that with him before. 

> Thanks for clarifying!

Thank you for bringing it up! It's never a bad idea to double check, and I really appreciate you did it 😉

**Comment by tyxla** [comment]
I'd also like to get a thumbs-up from @Automattic/team-calypso and confirm that they agree with what is being suggested in this PR.

**Comment by jeherve** [comment]
> I thought that the current - 1 version support was referring only to the WP Core versions supported by Jetpack when developing a feature and not to the Jetpack versions supported by Automattic (i.e. HEs). 

That was my understanding as well, I don't believe we've formalized the fact that we only support current Jetpack -1 anywhere until now. I'd be okay making this a rule though, that makes sense to me.

File: client/my-sites/post-selector/selector.jsx
Comment (jsnajdr) [inline client/my-sites/post-selector/selector.jsx]: This is interesting. The `PostSelector` (used, for example, when selecting a post to copy in Classic Editor UI) uses v1.2 of the `/posts` API, while other parts of Calypso, like the list of pages or posts, uses v1.1.

The v1.2 endpoint was added by @marekhrabe in #11887 and D4766-code to hide private post types from the selector.

Today, the v1.2 endpoint has been around for almost 3 years and we could use it by default everywhere? That's probably a change in some Post Query Manager in `state/posts`?

File: client/my-sites/comments/main.jsx
Comment (tyxla) [reply client/my-sites/comments/main.jsx]: Good catch 👍 Removed in https://github.com/Automattic/wp-calypso/pull/38585/commits/2546c5250affe484f56922d4f5731ccaff33d952.

File: client/my-sites/earn/memberships/index.jsx
Comment (tyxla) [reply client/my-sites/earn/memberships/index.jsx]: Of course 👍 Fixed in https://github.com/Automattic/wp-calypso/pull/38585/commits/a1b96522702f1c4f5486cfb1a6dd0cb7d8383cb8

File: client/my-sites/post-selector/selector.jsx
Comment (tyxla) [reply client/my-sites/post-selector/selector.jsx]: I'd be happy to do that, but it sounds like material for another PR 😉

**Comment by tyxla** [comment]
Thank you, everyone, for the great reviews!

I've rebased to the latest `master` and addressed all the feedback. This is ready for another 👀

**Review by jsnajdr — APPROVED** [review]
Thanks for addressing the review feedback! Looks good to :ship: if all the tests pass (or if we're sure the failures are unrelated). It would be nice to rebase and rerun CI before merging: the base `master` is very old here.

**Comment by tyxla** [comment]
Thanks, @jsnajdr! Rebased, will wait for CI and do some smoke testing in the meantime.

**Comment by tyxla** [comment]
Re-ran the tests and they passed. Smoke testing went well, I'm going to 🚢 

Thanks everyone for the feedback and reviews 🙌

**Pull request merged**
